From f94046db2243ae6bc89df9b8d98ff2dbab3423b2 Mon Sep 17 00:00:00 2001
From: Peter Colberg <peter.colberg@intel.com>
Date: Wed, 6 Dec 2023 16:40:27 -0500
Subject: [PATCH] fpga: dfl-pci-sva: depend on CONFIG_IOMMU_SVA

struct mm_struct only defines the member pasid when CONFIG_IOMMU_SVA
is enabled, e.g., by enabling CONFIG_INTEL_IOMMU_SVM.

Signed-off-by: Peter Colberg <peter.colberg@intel.com>
---
 drivers/fpga/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/fpga/Kconfig b/drivers/fpga/Kconfig
index f6b9499225e9..7031cb5485fb 100644
--- a/drivers/fpga/Kconfig
+++ b/drivers/fpga/Kconfig
@@ -237,7 +237,7 @@ config FPGA_DFL_PCI
 
 config FPGA_DFL_PCI_SVA
 	tristate "FPGA DFL PCIe Shared Virtual Addressing Device Driver"
-	depends on PCI && FPGA_DFL
+	depends on PCI && FPGA_DFL && IOMMU_SVA
 	help
 	  Select this option to enable PASID and IOMMU binding of DFL-based
 	  devices that support shared virtual memory.
